python/cugraph/cugraph/dask/sampling/uniform_neighbor_sample.py

```diff
@@ -123,47 +157,32 @@ def uniform_neighbor_sample(
     else:
         weight_t = "float32"
 
-    # start_list uses "external" vertex IDs, but if the graph has been
-    # renumbered, the start vertex IDs must also be renumbered.
+    if "_SRC_" in input_graph.edgelist.edgelist_df:
+        indices_t = input_graph.edgelist.edgelist_df["_SRC_"].dtype
+    elif src_n in input_graph.edgelist.edgelist_df:
+        indices_t = input_graph.edgelist.edgelist_df[src_n].dtype
+    else:
+        indices_t = numpy.int32
+
     if input_graph.renumbered:
         start_list = input_graph.lookup_internal_vertex_id(start_list).compute()
 
-    start_list = dask_cudf.from_cudf(
-        start_list, npartitions=min(input_graph._npartitions, len(start_list))
-    )
-    start_list = get_distributed_data(start_list)
-    wait(start_list)
-    start_list = start_list.worker_to_parts
-
     client = input_graph._client
 
-    result = [
-        client.submit(
-            _call_plc_uniform_neighbor_sample,
-            Comms.get_session_id(),
-            input_graph._plc_graph[w],
-            start_list[w][0],
-            fanout_vals,
-            with_replacement,
-            workers=[w],
-            allow_other_workers=False,
-        )
-        for w in Comms.get_workers()
-    ]
-
-    wait(result)
+    session_id = Comms.get_session_id()
 
-    cudf_result = [
-        client.submit(convert_to_cudf, cp_arrays, weight_t) for cp_arrays in result
-    ]
-
-    wait(cudf_result)
-
-    ddf = dask_cudf.from_delayed(cudf_result).persist()
+    # Send tasks all at once
```

Comment:
Just one comment; it would be nice to have a helper function for this so we can apply it to more algorithms.  But I suggest we leave that to a future PR given how much work we have left before burndown.